drivers/sensor/bmi270/bmi270_i2c.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Bus-specific functionality for BMI270s accessed via I2C.
+ */
+
+#include "bmi270.h"
+
+#if BMI270_BUS_I2C
```

Comment:
why not skip compilation of this file if not needed?

Reply:
> why not skip compilation of this file if not needed?

That's not possible to do with the current cmake/modules/extensions.cmake, and this is following the pattern for BME280. Unless you have some suggestion for how to do this with existing cmake support for DT?

Reply:
After a talk with @kapi-no he suggested a solution. Please check Kconfig file and CMakeLists.txt to see it.